Dockerfile.cpu

```diff
@@ -25,87 +24,44 @@ RUN apt-get update \
-RUN if [ "$TORCHVISION_PIP" ]; then ${CONDA_DIR}/bin/python3.6 -m pip install $TORCHVISION_PIP; fi
-RUN if [ "$TENSORPACK_PIP" ]; then ${CONDA_DIR}/bin/python3.6 -m pip install $TENSORPACK_PIP; fi
-RUN if [ "$LIGHTNING_PIP" ]; then ${CONDA_DIR}/bin/python3.6 -m pip install $LIGHTNING_PIP; fi
-
-RUN ${CONDA_DIR}/bin/python3.6 -m pip install determined determined-cli
+RUN if [ "$TENSORFLOW_PIP" ]; then python -m pip install $TENSORFLOW_PIP; fi
+RUN if [ "$TORCH_PIP" ]; then python -m pip install $TORCH_PIP; fi
+RUN if [ "$TORCHVISION_PIP" ]; then python -m pip install $TORCHVISION_PIP; fi
+RUN if [ "$LIGHTNING_PIP" ]; then python -m pip install $LIGHTNING_PIP; fi
 
 # We uninstall these packages after installing. This ensures that we can
 # successfully install these packages into containers running as non-root.
 # `pip` does not uninstall dependencies, so we still have all the dependencies
 # installed.
-RUN ${CONDA_DIR}/bin/python3.6 -m pip uninstall -y determined determined-cli determined-common
+RUN python -m pip install determined determined-cli
+RUN python -m pip uninstall -y determined determined-cli determined-common
+
+RUN python -m pip install -r /tmp/det_docker_files/notebook-requirements.txt
+ENV JUPYTER_CONFIG_DIR=/run/determined/jupyter/config
+ENV JUPYTER_DATA_DIR=/run/determined/jupyter/data
+ENV JUPYTER_RUNTIME_DIR=/run/determined/jupyter/runtime
 
-COPY files/additional-requirements.txt files/additional-requirements.txt
-RUN ${CONDA_DIR}/bin/python3.6 -m pip install -r files/additional-requirements.txt
+RUN python -m pip install -r /tmp/det_docker_files/additional-requirements.txt
 
-COPY files/notebook-requirements.txt files/notebook-requirements.txt
-RUN ${CONDA_DIR}/bin/python3.6 -m pip install -r files/notebook-requirements.txt
-ENV JUPYTER_CONFIG_DIR=/run/determined/jupyter/config JUPYTER_DATA_DIR=/run/determined/jupyter/data JUPYTER_RUNTIME_DIR=/run/determined/jupyter/runtime
+RUN /tmp/det_docker_files/add_det_nobody_user.sh
 
-# Add a det-nobody user. Unlike the traditional nobody user, det-nobody will
-# have a HOME directory that exists in order to support tools which require a
-# home directory, like gsutil. det-nobody is designed to be an out-of-the-box
-# solution for running workloads in unprivileged containers.
-RUN groupadd --gid 65533 det-nobody \
-  && useradd --create-home --home-dir /tmp/det-nobody --uid 65533 --gid 65533 --shell /bin/bash det-nobody
+RUN /tmp/det_docker_files/install_libnss_determined.sh
 
-# Add a plugin to the user system that lets us extend the users available in
-# the container at runtime. This is critical for supporting non-root shell,
-# which in turn is critical for non-root distributed training.
-COPY libnss_determined /tmp/libnss_determined
-RUN make -C /tmp/libnss_determined libnss_determined.so.2 install \
-  && rm -rf /tmp/libnss_determined \
-  && sed -E -i -e 's/^((passwd|shadow|group):.*)/\1 determined/' /etc/nsswitch.conf
+RUN /tmp/det_docker_files/cleanup.sh
```

Comment:
Honestly I'm not sure this one needs to get pulled out the way the others do.  There is the cache-size reduction which I think is pretty standard to run like `RUN apt instal ... && rm -rf /var/cache/apt` or something, and there are the deleting preconfigured ssh keys which is only required to pass the NGC security screen (even though those keys are totally unused in our system).